server/venueless/core/services/world.py

```diff
@@ -134,12 +139,24 @@ def _create_room(data, with_channel=False, permission_preset="public", creator=N
 
 
 async def create_room(world, data, creator):
-    for m in data.get("modules", []):
-        if m.get("type") != "chat.native":
-            raise ValidationError(
-                f"The dynamic creation of rooms with the module '"
-                f"{m['type']}' is currently not allowed."
-            )
+    types = {m['type'] for m in data.get("modules", [])}
+    if types == {'chat.native'}:
+        if not await world.has_permission_async(user=creator, permission=Permission.WORLD_ROOMS_CREATE_CHAT):
+            raise ValidationError("This user is not allowed to create a room of this type.")
```

Comment:
How do we communicate errors to the frontend? I'd low-key prefer an error code, at least in addition to the message.

Reply:
Yeah, you're right. Didn't want to raise ConsumerException right from the service, but it's probably the easiest way.